generator/src/main/java/com/scottlogic/deg/generator/validators/VisualisationConfigValidator.java

```diff
@@ -0,0 +1,51 @@
+package com.scottlogic.deg.generator.validators;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.outputs.targets.FileOutputTarget;
+import com.scottlogic.deg.generator.outputs.targets.OutputTarget;
+import com.scottlogic.deg.generator.utils.FileUtils;
+
+import java.util.ArrayList;
+
+/**
+ * Class used to determine whether the command line options are valid for visualisation
+ */
+public class VisualisationConfigValidator {
+
+    private final FileUtils fileUtils;
+
+    @Inject
+    public VisualisationConfigValidator(FileUtils fileUtils) {
+        this.fileUtils = fileUtils;
+    }
+
+    /**
+     * @return the result of command line validation. Contains a list of error messages.
+     * if the list is empty then the validation was successful.
+     */
+    public ValidationResult validateCommandLine(boolean overwite, OutputTarget outputTarget) {
+        return new ValidationResult(checkOutputTarget(overwite, outputTarget));
+    }
+
+    /**
+     * make sure the output file specified on the command line is valid. to be a valid output target file
+     * the target must not be a directory, and must not already exist.
+     *
+     * @param outputTarget the output target to check for validity.
+     * @return the list of error messages to append to if the output target is not valid.
+     */
+    private ArrayList<String> checkOutputTarget(boolean overwite, OutputTarget outputTarget) {
+        ArrayList<String> errorMessages = new ArrayList<>();
+        if (outputTarget instanceof FileOutputTarget) {
```

Comment:
You could exit early if `outputTarget` isn't a `FileOutputTarget` which could save some of the repeated hard-casts